src/tests/src/solver/optim/api/testApiOrtoolsLinearProblem.cpp

```diff
@@ -0,0 +1,242 @@
+{
+    pb->addIntVariable(0, 1, "a");
+    pb->addNumVariable(0, 1, "b");
+
+    pb->addConstraint(0, 1, "c");
+}
+
+BOOST_FIXTURE_TEST_CASE(linearProblemGetAndConstraintSetCoeff, Fixture)
+{
+    pb->addVariable(0, 1, true, "a");
+    pb->addConstraint(0, 1, "c");
+
+    auto* var = pb->getVariable("a");
+    auto* cons = pb->getConstraint("c");
+
+    BOOST_CHECK_EQUAL(var->getName(), "a");
+    BOOST_CHECK_EQUAL(cons->getName(), "c");
+
+    cons->setCoefficient(var, 3.2);
+    BOOST_CHECK_EQUAL(cons->getCoefficient(var), 3.2);
+
+    BOOST_CHECK_THROW(cons->setCoefficient(nullptr, 0), std::bad_cast);
+    BOOST_CHECK_THROW(cons->getCoefficient(nullptr), std::bad_cast);
+}
+
+bool correctMessage(const std::exception& ex)
+{
+    BOOST_CHECK_EQUAL(ex.what(), std::string("Element name already exists in linear problem"));
+    return true;
+}
+
+BOOST_FIXTURE_TEST_CASE(nameOrConstraintAlreadyExists, Fixture)
+{
+    pb->addNumVariable(0, 1, "a");
+    BOOST_CHECK_EXCEPTION(pb->addNumVariable(0, 1, "a"), std::exception, correctMessage);
+
+    pb->addConstraint(0, 1, "c");
+    BOOST_CHECK_EXCEPTION(pb->addConstraint(0, 1, "c"), std::exception, correctMessage);
+}
+
+BOOST_FIXTURE_TEST_CASE(getVarOrConstraintDoesntExist, Fixture)
+{
+    BOOST_CHECK_THROW(pb->getVariable(""), std::out_of_range);
+    BOOST_CHECK_THROW(pb->getConstraint(""), std::out_of_range);
+}
+
+BOOST_FIXTURE_TEST_CASE(maximizeMinimize, Fixture)
+{
+    pb->setMinimization();
+    BOOST_CHECK(pb->isMinimization());
+
+    pb->setMaximization();
+    BOOST_CHECK(pb->isMaximization());
+}
+
+BOOST_FIXTURE_TEST_CASE(mipVariableBounds, Fixture)
+{
+    auto* var = pb->addNumVariable(0, 1, "a");
+
+    var->setLb(-4);
+    var->setUb(7);
+
+    BOOST_CHECK_EQUAL(var->getLb(), -4);
+    BOOST_CHECK_EQUAL(var->getUb(), 7);
+
+    var->setBounds(2, 13);
+
+    BOOST_CHECK_EQUAL(var->getLb(), 2);
+    BOOST_CHECK_EQUAL(var->getUb(), 13);
+
+    auto* sameVar = pb->getVariable("a");
+    BOOST_CHECK_EQUAL(sameVar->getLb(), 2);
+    BOOST_CHECK_EQUAL(sameVar->getUb(), 13);
+}
+
+BOOST_FIXTURE_TEST_CASE(mipConstraintBounds, Fixture)
+{
+    auto* constraint = pb->addConstraint(0, 1, "a");
+
+    constraint->setLb(-4);
+    constraint->setUb(7);
+
+    BOOST_CHECK_EQUAL(constraint->getLb(), -4);
+    BOOST_CHECK_EQUAL(constraint->getUb(), 7);
+
+    constraint->setBounds(2, 13);
+
+    BOOST_CHECK_EQUAL(constraint->getLb(), 2);
+    BOOST_CHECK_EQUAL(constraint->getUb(), 13);
+}
+
+BOOST_FIXTURE_TEST_CASE(objectiveCoeff, Fixture)
+{
+    auto* var = pb->addVariable(0, 1, true, "a");
+    pb->setObjectiveCoefficient(var, 1);
+    BOOST_CHECK_EQUAL(pb->getObjectiveCoefficient(var), 1);
+
+    BOOST_CHECK_THROW(pb->getObjectiveCoefficient(nullptr), std::bad_cast);
+    BOOST_CHECK_THROW(pb->setObjectiveCoefficient(nullptr, 0), std::bad_cast);
+}
+
+BOOST_FIXTURE_TEST_CASE(infeasibleProblem, Fixture)
+{
+    auto* a = pb->addIntVariable(0, 10, "a");
```

Comment:
If we want to test only how the problem reacts to resolution when it's infeasible, then we could use a even more simple problem : one variable (always positive) and one constraint (always negative), not even sure that we need a cost for variables.
This will make the test even shorter.

Reply:
Done in #2377 and merged here